cheroot/workers/threadpool.py

```diff
@@ -180,8 +183,16 @@ def put(self, obj):
                 waiting to be processed
         """
         self._queue.put(obj, block=True, timeout=self._queue_put_timeout)
-        if obj is _SHUTDOWNREQUEST:
-            return
+
+    def _clear_dead_threads(self):
+        # Remove any dead threads from our list
+        for t in self._threads:
```

Comment:
Alternatively, just filter out stuff right here:
```suggestion
        for t in (t for t in self._threads if t.isAlive()):
```

Reply:
Well, you meant `if not t.isAlive()`, but I've incorporated that change.